### PR TITLE
AutoHeuristic: tests

### PR DESCRIPTION
**Pull request by AlnisM**
This PR adds tests to AutoHeuristic that ensure that when existing heuristics are re-generated, the generated code stays the same.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #133496

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: torchgen/_autoheuristic/train_regression.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ def main(self, log_path, other_datasets, nrows, heuristic_name):
         self.add_real_datasets(datasets, other_datasets, cat_feature2cats)
 
         # We will do a grid search over the values
-        max_depths = [5, 10, 13, 15, 17, 20, 23, None]
+        max_depths = [5, 10]
```
Comment (eellison) [inline torchgen/_autoheuristic/train_regression.py:49]: why were these changed ? what does this entail ?

**Review by eellison — APPROVED** [review]
great to have tests of regeneration, thank you!

File: torchgen/_autoheuristic/train_regression.py (hunk 1 above)
Comment (AlnisM) [reply torchgen/_autoheuristic/train_regression.py:49]: We never want to use max_depths this large (the decision trees and the generated code will be huge), so I removed them, because trying out many different hyperparameters increases the time to learn the heuristic a lot. train_regression.py is only used for pad_mm, where a max_depth of 5 is the best choice, so it doesn't change anything for existing heuristics. I'll actually change it to max_depths = [5, 6, 7], because we probably dont want trees with max_depth > 7 and from my experience max_depth < 5 is not enough to learn a good heuristic.

Comment (eellison) [reply torchgen/_autoheuristic/train_regression.py:49]: add this comment inline ?
